### PR TITLE
New app shell 🐚 

### PR DESCRIPTION
**Pull request by pvinis**
The type of this PR is: **TYPE**

This PR resolves []

### PR Checklist (tick all before merging)

- [ ] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [ ] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [ ] I have documented any follow-up work that this PR will require, or it does not require any.
- [ ] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [ ] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -1,3 +1,4 @@
 nodejs 14.17.1
 ruby 2.7.2
 python 3.7.12
+java adoptopenjdk-openj9-15.0.2+7.openj9-0.24.0
```
Comment (pvinis) [inline .tool-versions:4]: apparently java 16 doesnt work yet, so we keep 15.

File: android/app/build.gradle [hunk 2]
```diff
@@ -127,15 +127,17 @@ def jscFlavor = 'org.webkit:android-jsc-intl:+'
  */
 def enableHermes = project.ext.react.get("enableHermes", false);
 
+/**
```
Comment (pvinis) [inline android/app/build.gradle:130]: most changes here are from 0.63 -> 0.66 upgrade

File: android/app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -99,8 +99,6 @@
             </intent-filter>
         </activity>
 
-        <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />
```
Comment (pvinis) [inline android/app/src/main/AndroidManifest.xml:102]: from 0.63 -> 0.66 upgrade
moved to the debug xml file.

File: android/app/src/debug/AndroidManifest.xml [hunk 4]
```diff
@@ -4,5 +4,10 @@
 
     <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
 
-    <application android:usesCleartextTraffic="true" tools:targetApi="28" tools:ignore="GoogleAppIndexingWarning" />
+    <application
+        android:usesCleartextTraffic="true"
+        tools:targetApi="28"
+        tools:ignore="GoogleAppIndexingWarning">
+        <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />
```
Comment (pvinis) [inline android/app/src/debug/AndroidManifest.xml:11]: look in the file below

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c0687c5987d51b74fb5301ed75a31742c3065db4
</p>

File: android/build.gradle [hunk 5]
```diff
@@ -10,13 +10,15 @@ buildscript {
         targetSdkVersion = 30
         firebaseIidVersion = "21.1.0" // Needed for react-native-device-info
         googlePlayServicesAuthVersion = "16.0.1"
+        ndkVersion = "21.4.7075529"
```
Comment (pvinis) [inline android/build.gradle:13]: most changes here are from 0.63 -> 0.66 upgrade

File: android/build.gradle [hunk 6]
```diff
@@ -36,8 +39,18 @@ allprojects {
         }
 
         google()
-        jcenter()
         maven { url 'https://www.jitpack.io' }
         maven { url "http://appboy.github.io/appboy-android-sdk/sdk" }
+        jcenter()
+        maven {
```
Comment (pvinis) [inline android/build.gradle:45]: mapbox being difficult on android too

File: android/gradlew.bat [hunk 7]
```diff
@@ -40,7 +40,7 @@ if defined JAVA_HOME goto findJavaFromJavaHome
 
 set JAVA_EXE=java.exe
 %JAVA_EXE% -version >NUL 2>&1
-if "%ERRORLEVEL%" == "0" goto init
+if "%ERRORLEVEL%" == "0" goto execute
```
Comment (pvinis) [inline android/gradlew.bat:43]: most changes here are from 0.63 -> 0.66 upgrade

File: patches/react-native+0.66.4.patch [hunk 8]
```diff
@@ -0,0 +1,75 @@
+diff --git a/node_modules/react-native/Libraries/Core/Devtools/parseErrorStack.js b/node_modules/react-native/Libraries/Core/Devtools/parseErrorStack.js
+index 185ddcb..113671f 100644
+--- a/node_modules/react-native/Libraries/Core/Devtools/parseErrorStack.js
++++ b/node_modules/react-native/Libraries/Core/Devtools/parseErrorStack.js
+@@ -13,6 +13,7 @@
+ import type {StackFrame} from '../NativeExceptionsManager';
+ import type {HermesParsedStack} from './parseHermesStack';
+ 
++const stacktraceParser = require('stacktrace-parser');
```
Comment (pvinis) [inline patches/react-native+0.66.4.patch:9]: still the same

File: patches/react-native+0.66.4.patch [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+     : global.HermesInternal
+diff --git a/node_modules/react-native/Libraries/LayoutAnimation/LayoutAnimation.js b/node_modules/react-native/Libraries/LayoutAnimation/LayoutAnimation.js
+index efb12af..09b5978 100644
+--- a/node_modules/react-native/Libraries/LayoutAnimation/LayoutAnimation.js
++++ b/node_modules/react-native/Libraries/LayoutAnimation/LayoutAnimation.js
+@@ -64,9 +64,9 @@ function configureNext(
+ 
+   // In Fabric, LayoutAnimations are unconditionally enabled for Android, and
+   // conditionally enabled on iOS (pending fully shipping; this is a temporary state).
+-  const FabricUIManager: FabricUIManagerSpec = global?.nativeFabricUIManager;
```
Comment (pvinis) [inline patches/react-native+0.66.4.patch:29]: some weird issue with LayoutAnimation

File: Podfile [hunk 10]
```diff
@@ -214,8 +214,8 @@ post_install do |installer|
   # * Send PRs for the rest
   %w[
     Pods/ORStackView/Classes/ios/ORStackView.h
-    Pods/NAMapKit/NAMapKit/*.h
   ].flat_map { |x| Dir.glob(x) }.each do |header|
+    system("chmod +w #{header}")
```
Comment (pvinis) [inline Podfile:218]: these are needed because the files are read-only when we get them from cocoapods

File: patches/react-native+0.66.4.patch [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+     NSInteger touchID = ([_reactTouches.lastObject[@"identifier"] integerValue] + 1) % RCTMaxTouches;
+diff --git a/node_modules/react-native/scripts/find-node.sh b/node_modules/react-native/scripts/find-node.sh
+index c498ba6..6a65757 100755
+--- a/node_modules/react-native/scripts/find-node.sh
++++ b/node_modules/react-native/scripts/find-node.sh
+@@ -12,6 +12,15 @@ if [[ -d $HOMEBREW_M1_BIN && ! $PATH =~ $HOMEBREW_M1_BIN ]]; then
+   export PATH="$HOMEBREW_M1_BIN:$PATH"
+ fi
+ 
++# Set up asdf-vm if present
```
Comment (pvinis) [inline patches/react-native+0.66.4.patch:84]: asdf 💜

File: ios/eigen/NativeModules/ARTArtsyNativeModule.h [hunk 12]
```diff
@@ -0,0 +1,10 @@
+#import <Foundation/Foundation.h>
+#import <React/RCTBridgeModule.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface ARTArtsyNativeModule : NSObject <RCTBridgeModule>
```
Comment (pvinis) [inline ios/eigen/NativeModules/ARTArtsyNativeModule.h:6]: finally, we have the `ART` prefix in. 💜

**Pull request closed without merging**